### PR TITLE
Restructure to registry-less ECS

### LINKED ISSUE #74: Abstract GL resource compilation away from systems
Right now, `SysVehicle::part_instantiate` has code that attempts to get the resources required by the vehicle, and if they don't yet exist, attempts to compile them from image/mesh data in resource packs. This pattern is repeated consistently in several places in the repo, and will likely continue to be common. We may therefore want to consider abstracting this resource check away, so that systems can call a single function to fetch the required data without having to manually check for and/or compile the GL resources every time.

### PR DESCRIPTION
**Pull request by Capital-Asterisk**
There are many casualties, but lots of cool things as a result.

Depends on #157 

**Almost everything is disabled and many parts of the code remain unused. Only a few files are important to review.**

Many files have an `#if 0` block covering the whole thing. Please ignore those files.

## Problem

The entt::registry is designed for the common case, but had started to become a burden as the project's architecture evolved.

Consider that the game scene (ActiveScene) needs to be synchronized to a Universe, associated with resources tied to the GPU, and access data shared between scenes. It may make sense to store references to each of these external dependencies, but instead, it's far more practical (in the case of ECS) to pass them all as function parameters. The application that calls all the functions/systems will have easy access to each of the dependencies. The case right now is that the systems themselves service-locate, which unneeded complexity and is harder to control from outside.

For example, if we want to manage thread-local instances of a component pool, dirty flags, or queues, it would be really easy to handle them at the application-level and pass them to system functions as parameters. The systems themselves needing to locate the thread-local instances would be a headache in comparison.

The best system function is one that has very little to no side effects, and only modifies whatever mutable parameters its given. This also means it's best to pass individual component pools and views as parameters, instead of the entire scene each time. This makes it obvious which components a system modifies, as they can be marked const too, and makes it completely independent of the entt::registry. In my opinion, passing around a huge centralized class is an anti-pattern nonetheless.

Other problems with the registry:
* It's used to store _everything_ because it's more convenient to access
* Things that didn't need to be components were made into components anyways
* Many parts of the project are already designed to work around the registry's shortcomings. ie. Coordinate spaces

I'm confident that the root cause to many of these problems are because of the registry.

## Changes

Many parts of the code were touched, mostly to replace ActiveScene from their arguments to individual views and storages. There were changes to things related to universe, machines, and vehicles, but none of them are used yet.

All the broken systems will be reworked in following PRs.

### No more centralized scene state

RIP ActiveScene

"Scene" now refers to any structure that stores the state of a game world (with entities and components).

Instead of using the entt::registry, components are stored as entt::basic_storages. For example, see [basic.h](https://github.com/Capital-Asterisk/osp-magnum/blob/no-registry/src/osp/Active/basic.h); acomp_storage_t is typedefed in [activetypes.h](https://github.com/Capital-Asterisk/osp-magnum/blob/no-registry/src/osp/Active/activetypes.h).

System functions can accept views to read or modify components, or take a reference to the storage to be able to create or destroy components.

Collecting together all the component storages into some structure (such as a struct) is up to the application. The system functions won't be aware of how the component storages are stored. IdRegistry can be used to generate entity IDs.

The Universe class is not dead, since it's responsible for storing CoordinateSpaces.

### Simplified Render system

Lots of complexity was turned into simple functions. Many structures didn't need to be tied to the scene, or didn't need to exist at all. The GPU functions that get called are still the same, but are less tangled and far easier to follow. Additionally, OpenGL-specific code from SysRender was split into SysRenderGL, cleanly separating it from any scene logic.

This fixes #74

### Revamped Test Application - The important part

A small self-contained 'Engine Test' is added to demonstrate osp-magnum's use as a game engine. See [scenarios_enginetest.cpp](https://github.com/Capital-Asterisk/osp-magnum/blob/no-registry/src/test_application/activescenes/scenarios_enginetest.cpp). All it does right now is display a spinning cube.

The renderer state is now entirely separated from the scene logic. The Magnum application (ActiveApplication) can be closed and re-opened without needing to destruct the scene. This will also make it incredibly easy to run a scene in the background.

I'd highly recommend reading through `scenarios_enginetest.cpp` more thoroughly, as it should give a good understanding of the new API.

### main.cpp rework

This partially addresses #155

Lots of things were removed; again, no universe stuff is functional. A map of commands for loading scenes was added. Scenes are now created separately from the Magnum thread, as they are no longer coupled to gpu resources.

### CODE REVIEW

File: src/test_application/activescenes/scenarios_physicstest.cpp [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    rScene.m_drawing.m_diffuseDirty.assign(std::begin(rMeshSet), std::end(rMeshSet));
+
+    return [&rScene, pRenderer = std::move(pRenderer)] (ActiveApplication& rApp)
+    {
+        update_test_scene(rScene);
+        render_test_scene(rApp, rScene, *pRenderer);
+    };
+}
+
+} // namespace testapp::physicstest
```
Comment (Capital-Asterisk) [inline src/test_application/activescenes/scenarios_physicstest.cpp:358]: This physics test scene is currently incomplete; ignore this entire file.

File: src/test_application/activescenes/scenarios_enginetest.cpp [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+constexpr int const gc_maxMaterials = 2;
+
+/**
+ * @brief State of the entire engine test scene
+ */
+struct EngineTestScene
+{
+    // ID registry generates entity IDs, and keeps track of which ones exist
+    osp::IdRegistry<osp::active::ActiveEnt> m_activeIds;
```
Comment (z-adams) [inline src/test_application/activescenes/scenarios_enginetest.cpp:72]: Does this actually completely replace the EnTT entity system? I suppose most of the ent-related functions we're used to are members of the component pools. Does this change preclude us from using functions like `reg.all_of<T...>(entity)` to check for component existence, or `reg.valid(entity)`? Are there replacements for these functionalities?

**Review by z-adams — APPROVED** [review]
Much cleaner, I like it.

File: src/test_application/activescenes/scenarios_enginetest.cpp (hunk 2 above)
Comment (Capital-Asterisk) [reply src/test_application/activescenes/scenarios_enginetest.cpp:72]: As far as I'm concerned, this works as a more capable replacement. Memory usage is lower, more efficient recycling of deleted entity IDs, but the main benefits come from the flexibility of manually managing component pools. The only thing we lose is the existing automation previously provided by the registry (and also entity versions, which we can implement if we really want to). From working on other branches, passing component pools as function parameters eliminates the need for many of the convenient functions in the first place.

`reg.all_of<>` on the inside is calling contains() on each sparse set in the registry. There aren't any instances of all_of being called with any more than a single component in the codebase. `reg.valid(ent)` is roughly equivalent to `m_activeIds.exists(ent)`.

**Pull request merged**